### DISCUSSION TITLE
Deploy a single-tenant (Standard Tier) Logic App without data loss

### DISCUSSION THREAD

**Discussion by DavidFuchs** [Azure/logicapps#768] [Q&A]
I'm deploying a logic app via Azure DevOps as a ZIP deployment. Is there any documentation detailing what happens to any existing workflow runs that are running when a new package for a Logic App is deployed?

I'm concerned about data loss. Are existing workflows killed/interrupted?  If so, are there practices that can be followed to ensure that no data loss occurs?

**Comment by efficient-soil** [comment]
As this question is still open I might add my take on this:

Stateful workflows are persisted at some state within the running/in-flight instance when the logic app is stopped. No data loss. But:

At several occasions we've had workflows persist state when the logic app is shutting down, then the deployment (Azure DevOps Pipeline) finishes and when the logic app starts again it resumes workflows from their persisted state, which in these cases mean that **actions in the workflow, that are depending on output from previous actions that were added in the deploy, fails**.

The only solution to this problem I've seen is to enable **Deployment Slots** but that is a whole different story since that requires us to create a complete new deployment slot environment. Or do we..?

I've been experimenting with disabling workflows on deploy which inactivates the trigger and then letting the in-flight processes finish/drain before the actual deploy starts. That makes the yaml very complex, with API request to Azure and handling of the responses in different ways.

If anyone have a solution or could clarify the deployment slot usage I would appreciate that. Can't really find anything about this related to Logic Apps.
